generators/generator-base-blueprint.js

```diff
@@ -84,51 +90,187 @@ module.exports = class extends BaseGenerator {
+            this.jhipsterConfig.blueprints = blueprints;
+            this.jhipsterConfig.otherModules = otherModules;
+
+            if (!this.options.skipChecks) {
+                const namespaces = blueprints.map(bp => jhipsterUtils.packageNameToNamespace(bp.name));
+                // Verify if the blueprints has been registered.
+                const missing = namespaces.filter(namespace => !this.env.isPackageRegistered(namespace));
+                if (missing && missing.length > 0) {
+                    this.error(`Some blueprints were not found ${missing}, you should install them manually`);
                 }
-                const blueprintGenerator = this.composeBlueprint(blueprint.name, subGen, bpOptions);
+            }
+        }
+
+        const blueprints = this.jhipsterConfig.blueprints;
+        if (blueprints && blueprints.length > 0) {
+            blueprints.forEach(blueprint => {
+                const blueprintGenerator = this.composeBlueprint(blueprint.name, subGen, extraOptions);
                 // If the blueprints sets sbsBlueprint property, then don't ignore the normal workflow.
-                if (!useBlueprints && blueprintGenerator && !blueprintGenerator.sbsBlueprint) {
+                if (blueprintGenerator && !blueprintGenerator.sbsBlueprint) {
                     useBlueprints = true;
                 }
             });
         }
         return useBlueprints;
     }
+
+    /**
+     * Compose external blueprint module
+     * @param {string} blueprint - name of the blueprint
+     * @param {string} subGen - sub generator
+     * @param {any} options - options to pass to blueprint generator
+     */
+    composeBlueprint(blueprint, subGen, extraOptions = {}) {
+        blueprint = jhipsterUtils.normalizeBlueprintName(blueprint);
+        if (!this.configOptions.skipChecks && !this.options.skipChecks) {
+            this.checkBlueprint(blueprint);
+        }
+
+        const generatorName = jhipsterUtils.packageNameToNamespace(blueprint);
+        const generatorNamespace = `${generatorName}:${subGen}`;
+        if (!this.env.isPackageRegistered(generatorName)) {
+            this.env.lookup({ filterPaths: true, packagePatterns: blueprint });
+        }
+        if (!this.env.get(generatorNamespace)) {
+            this.debug(
+                `No blueprint found for blueprint ${chalk.yellow(blueprint)} and ${chalk.yellow(
+                    subGen
+                )} subgenerator: falling back to default generator`
+            );
+            return false;
+        }
+
+        const finalOptions = {
+            ...this.options,
+            configOptions: this.configOptions,
+            ...extraOptions,
+            jhipsterContext: this,
+        };
+
+        const blueprintGenerator = this.composeWith(generatorNamespace, finalOptions, true);
+        if (blueprintGenerator instanceof Error) {
+            throw blueprintGenerator;
+        }
```

Comment:
I don't get it. A generator is a generator, right? why would something called `composeWith` **return** an Error?

Reply:
This is a legacy behaviour of yeoman-generator.
Nothing I can do for yeoman-generator 4.x.
If you want to create an issue at yeoman-generator repository for 5.x, I can handle it.